packages/js/msgpack/src/__tests__/msgpack.spec.ts

```diff
@@ -26,8 +26,17 @@ describe("msgpack", () => {
     customMap.set("secondKey", "secondValue")
 
     const encoded = msgpackEncode(customMap)
-    // [199, 43, 1] are being added because of the map structure
-    expect(encoded).toEqual(Uint8Array.from([ 199, 43, 1, ...expectedArrayLike]))
+    const decoded = msgpackDecode(encoded)
+    expect(decoded).toEqual(customMap);
+  });
+
+  it("Should encode and decode nested map, returning the same map", () => {
+    const customMap: Map<string, Map<string, string>> = new Map()
+    // @ts-ignore
+    customMap.set("firstKey", new Map([["one", "1"]]));
+    customMap.set("secondKey", new Map([["second", "2"]]));
+
+    const encoded = msgpackEncode(customMap)
     const decoded = msgpackDecode(encoded)
```

Comment:
`decodedMap` instead of `decoded` ?

Reply:
Yes we expect decodedMap but if the decoding is wrong then it might not be map.